src/formatting/formattable.rs

```diff
@@ -102,6 +105,60 @@ where
 // endregion custom formats
 
 // region: well-known formats
+impl sealed::Sealed for Rfc2822 {
+    fn format_into(
+        &self,
+        output: &mut impl io::Write,
+        date: Option<Date>,
+        time: Option<Time>,
+        offset: Option<UtcOffset>,
+    ) -> Result<usize, error::Format> {
+        let date = date.ok_or(error::Format::InsufficientTypeInformation)?;
+        let time = time.ok_or(error::Format::InsufficientTypeInformation)?;
+        let offset = offset.ok_or(error::Format::InsufficientTypeInformation)?;
+
+        let mut bytes = 0;
+
+        let year = date.year();
+
+        if year < 1900 {
+            return Err(error::Format::InvalidComponent("year"));
+        }
+        if time.nanosecond() != 0 {
+            return Err(error::Format::InvalidComponent("nanosecond"));
```

Comment:
Not sure whether to fail on non-zero nanoseconds or not, RFC2822 can't represent non-zero nanoseconds, but failing formatting on those seems like big potential for a footgun that would take library consumers by surprise. E.g. `OffsetDateTime::now_utc().format(&Rfc2822)?` would error in 99.999% of cases.

Reply:
I think it is reasonable to ignore if nonzero. That's what happens with manually created formats unless it's explicitly provided, so it would be consistent.

Reply:
Will fixup e257222 into 92d1a69 in that case